src/sql/src/plan/scope.rs

```diff
@@ -332,9 +339,7 @@ impl Scope {
         let table_name = None;
         self.resolve_internal(
             outer_scopes,
-            |_column, _lat_level, item| {
-                item.allow_unqualified_references && item.column_name == *column_name
-            },
+            |_column, _lat_level, item| item.allow_unqualified_references,
```

Comment:
Similarly, it would be nice to get a test case if possible for when this returns false.

Reply:
I'm going to set this to automerge, more than happy to come back and update the tests though!